### PR TITLE
Remove python 2 support

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
This PR removes support for Python 2. Once this is merged, pyOptSparse will no longer work with Python 2.
The main change was just removing the `six` package and changing the syntax to use the native py3 iterators.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- Breaking change (non-backwards-compatible fix or feature)

## Testing
All existing tests pass on py3.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,19 +11,19 @@ services:
...
-    - DOCKER_TAG=u18-gcc-ompi-py2-stable
-    - DOCKER_TAG=u18-gcc-ompi-py3-latest
-    - DOCKER_TAG=u18-gcc-ompi-py3-stable
-    - DOCKER_TAG=u20-gcc-ompi-py3-latest
-    - DOCKER_TAG=u20-gcc-ompi-py3-stable
+    # - DOCKER_TAG=c7-gcc-ompi-latest
+    - DOCKER_TAG=c7-intel-impi-latest
+    - DOCKER_TAG=u18-gcc-ompi-stable
+    - DOCKER_TAG=u18-gcc-ompi-latest
+    - DOCKER_TAG=u18-gcc-ompi-stable
```
Comment (marcomangano) [inline .travis.yml:24]: Since we removed the py2 test, I feel we have a duplicate `u18-gcc-ompi-stable` build here

File: .travis.yml [hunk 2]
```diff
@@ -11,19 +11,19 @@ services:
   - docker
 env:
   global:
-    - REPO_NAME=pyoptsparse
+    - REPO_NAME=$(eval basename $TRAVIS_REPO_SLUG)
```
Comment (marcomangano) [inline .travis.yml:14]: Just a curiosity here, how does this command work? what kind of env variable is `$TRAVIS_REPO_SLUG`?

**Review by marcomangano — COMMENTED** [review]
Check the comment about the extra tag below, looks good otherwise

File: .travis.yml (hunk 1 above)
Comment (nwu63) [reply .travis.yml:24]: Yes very good catch.

File: .travis.yml (hunk 2 above)
Comment (nwu63) [reply .travis.yml:14]: It's set to `mdolab/pyoptsparse` for this repo. `basename` picks out the last portion after all the `/` so it's a convenient way to get the repo name programmatically.

**Review by marcomangano — APPROVED** [review]

**Review by sseraj — APPROVED** [review]

**Pull request merged**